butlarr/services/bazarr.py

```diff
@@ -0,0 +1,313 @@
+        rows_action = []
+
+        parent = state.parent.service
+        parent_menu = state.parent.menu
+        parent_state = state.parent.state
+        
+        if state.menu == 'list':
+            if len(state.items) > 0:
+                row_navigation = [Button("=== Subtitles ===", "noop")]
+            else:
+                row_navigation = [Button("=== No subtitles found ===", "noop")]
+
+            current_index = 0
+            for item in state.items:
+                description = f"[Score: {item['score']}]"
+                description += f" {item['release_info'][0]}"
+                description = description[0:1024]
+                rows_menu.append(
+                    [
+                        Button(
+                            description,
+                            self.get_clbk("download", state.index),
+                        ),
+                    ]
+                )
+                current_index = current_index + 1
+                if current_index >= 5: break
+        
+        if state.menu == 'success':
+            row_navigation = [Button("Subtitle downloaded!", "noop")]
+        if parent_menu:
+            rows_action.append([Button("🔙 Back", parent.get_clbk(parent_menu))])
+        elif parent_state.menu:
+            rows_action.append([Button("🔙 Back", parent.get_clbk(parent_state.menu))])
+        
+
+        return [row_navigation, *rows_menu, *rows_action]
+
+
+    def detect_api(self, api_host):
+        status = None
+        # Detect version and api_url
+        try:
+            self.api_url = f"{api_host.rstrip('/')}/api"
+            status = self.request("system/status")
+        finally:
+            if status is None:
+                logger.error(
+                    "Could not reach compatible api. Is the service down? Is your API key correct?"
+                )
+                exit(1)
+            assert (
+                status
+            ), "Could not reach compatible api. Is the service down? Is your API key correct?"
+            api_version = status.get("data", {}).get("bazarr_version", "")
+            assert api_version, "Could not find compatible api."
+            return api_version
+    
+    
+    def search(self, arr_variant, id):
+        if arr_variant == ArrVariant.RADARR:
+            status = self.request('providers/movies', params={'radarrid': id}, fallback=[])
+            return status.get('data') if len(status) > 0 else status
+        if arr_variant == ArrVariant.SONARR:
+            status = self.request('providers/episodes', params={'episodeid': id}, fallback=[])
+            return status.get('data') if len(status) > 0 else status
+        else:
+            assert False, f"Bazarr integration not Implemented"
+
+    def download(
+        self,
+        service,
+        id,
+        item=None,
+        options={},
+    ):
+        assert item, "Missing required arg! You need to provide a item!"
+
+        if service == ArrVariant.RADARR:
```

Comment:
The service should provide functions for these behaviours.
You already got a Addon class, use it to force an interface for that :)

Reply:
I'm not sure I follow here. 

The idea here is that the bazarr Service must provide all the functionality to integrate with bazarr and sonarr, because that's what bazarr, the application, do: You can't use bazarr as a standalone service. 

The addon class I created is used for services that can be used as addons (bazarr), not for services that have addons (radarr and sonarr). 

Of course we can change that, just not sure how you would do it though.

Reply:
No I dont' think we should change that, your approach is the correct way imo.
We don't want to start moving "addon functionality" into the sonarr and radarr implementations, it might just cause confusion down the line